fe/src/main/java/org/apache/doris/load/routineload/RoutineLoad.java

```diff
@@ -251,6 +355,31 @@ public void updateRoutineLoadJobState(RoutineLoadJob routineLoadJob, RoutineLoad
         }
     }
 
+    public void processTimeOutTasks() {
+        List<RoutineLoadTaskInfo> runningTasks = new ArrayList<>(idToRoutineLoadTask.values());
+        runningTasks.removeAll(needSchedulerRoutineLoadTask);
+
+        for (RoutineLoadTaskInfo routineLoadTaskInfo : runningTasks) {
+            routineLoadTaskInfo.writeLock();
```

Comment:
It's highly recommended NOT to expose lock outside a class.
It will cause a lot of troubles.

Reply:
And I can't see why you need a lock here?
Nothing need to be protected in RoutineLoadTaskInfo?

Reply:
‘idToRoutineLoadTask’  is the member of RoutineLoad, it should not be protected by lock in RoutineLoadTaskInfo.

Reply:
In the logical, every task only can process one function, either `processTimeOutTask` or `commit task`. So it really need a segment lock for `idToRoutineLoadTask`. According to the lot of time of `commit task`, I will use one lock instead of per task per lock.